include/fpcsc/perf_util/sleep.h

```diff
@@ -12,15 +12,26 @@ inline void sleep_for_secs(unsigned Secs) {
   std::this_thread::sleep_for(std::chrono::seconds(Secs));
 }
 
+inline void busy_sleep_for_secs(unsigned Secs) {
+  std::cout << "Sleeping for " << Secs << " seconds" << std::endl;
+  auto start_ms = std::chrono::duration_cast<std::chrono::milliseconds>(
+      std::chrono::high_resolution_clock::now().time_since_epoch());
+  auto end_ms = start_ms + std::chrono::seconds(Secs);
+  auto current_ms = start_ms;
+
+  while (current_ms < end_ms) {
+    for (long counter = 0; counter < 100'000'000; ++counter) {
+      asm volatile("" : "+g"(counter) : :); // prevent optimization
+    }
+    current_ms = std::chrono::duration_cast<std::chrono::milliseconds>(
+      std::chrono::high_resolution_clock::now().time_since_epoch());
+  }
+}
+
 inline void sleep_for_millisecs(unsigned Millisecs) {
   std::cout << "Sleeping for " << Millisecs << " milliseconds" << std::endl;
   std::this_thread::sleep_for(std::chrono::milliseconds(Millisecs));
 }
-
-inline void sleep_for_nanosecs(unsigned millisecs) {
-  std::this_thread::sleep_for(std::chrono::nanoseconds(millisecs));
-}
```

Comment:
Just, keep the function, maybe someone else will need it.

Reply:
In SimpleSleepLoop, this lead to a lot of variance. I would rather force whoever plans to use it to carefully consider whether sleeping for a few nanoseconds is actually necessary as this corresponds to only a few hundred processor cycles. In my opinion, it makes sense to use a busy sleep version instead if necessary.

Reply:
A busy version in addition would sure be nice. The core point about this is also not that it's super precise but more that one could, with the right experiment design, try to analyze implications to sampling resolution.
Now that I think about it, as perf does not find the sleep for this a busy version would be really practicable.

Reply:
Let's remove it for now and add one if we need it